Agents/OSMAgent/osmagent/src/main/resources/FeatureInfoAgent/queries/building_usage.sparql

```diff
@@ -0,0 +1,28 @@
+PREFIX obe: <https://www.theworldavatar.com/kg/ontobuiltenv/>
+PREFIX om: <http://www.ontology-of-units-of-measure.org/resource/om-2/>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+
+SELECT DISTINCT ?label ?value ?unit
+WHERE {
+SERVICE [ONTOP] {
+SELECT DISTINCT ?label ?value ?unit
+WHERE {
+            {
+
+                [IRI] obe:hasPropertyUsage ?usage .
+                ?usage rdf:type ?value_ini .
+                OPTIONAL { ?usage obe:hasUsageShare ?unit_ini . }
+                BIND (replace(str(?value_ini), str(obe:), '') as ?usage_type)
```

Comment:
While I agree in principle, I believe your comment is misplaced here: In case you refer to the `BIND` clause, this simply tidies up the UsageType for the visualisation, i.e., removes the IRI prefix from the term shown in the side panel

Reply:
You have just confirmed why this is criminal: The `BIND` clause takes the `rdf:type` of `?usage`, which is an IRI, then strips out the `obe` namespace part of the string, and then puts the remaining string into the `?usage_type` variable (which is then integrated into the label string to be displayed by the visualisation). How is that not inferring information from an IRI string?!?

Reply:
I thought as this is for visualisation/humans only this is ok based on our current default approach to create IRIs, which is based on human readable concept names. But I do understand your concern.
So, would uploading the ontolgy into the same namespace and then asking for `?usage rdf:type/rdfs:label ?value_ini` resolve this issue from your persepective?

Reply:
Thank you for your comments Seb. 

The current time for FIA to query takes 3 seconds. This invovles stripping the IRI <https://www.theworldavatar.com/kg/ontobuiltenv/Clinic> and retains "Clinic" as the usage, simply to tidy up the visualisation instead of leaving the whole IRI to be displayed in TWA-VF. 

Attached below is the SPARQL query we attempted: 
1) Uploading to the OntoBuiltEnv.owl file onto BG - [building_usage_with_OWLfile.txt](https://github.com/cambridge-cares/TheWorldAvatar/files/12698191/building_usage_with_OWLfile.txt) takes about 30 seconds for FIA to query. 
2) Without uploading the OntoBuiltEnv.owl - [building_usage_without_OWL.txt](https://github.com/cambridge-cares/TheWorldAvatar/files/12698192/building_usage_without_OWL.txt) takes about 40 seconds for FIA to query.

The processing time might be too long for the metadata to be retrieved and shown on TWA-VF. 

Let us know what should our next step be or if there could be a faster way to do it.

Reply:
Hard-coding the usage labels into the query string is quite bad, too (about as bad as extracting them from IRI strings, I would say).

Representing the usage labels as triples in Blazegraph seems like the right thing to do, in principle. I don't understand, though, why you seem to be federating that sub-query to Ontop, if the triples are in Blazegraph? Unless you uploaded the usage types also into PostGIS, I'm not even sure how that can work...

Reply:
In building_usage_with_OWLfile.txt, by uploading the OntoBuiltEnv.owl file, 
`?usagetypeiri rdfs:label ?label`  retrieves the label from the OntoBuiltenv.owl in BG. The inner query `{[IRI] obe:hasPropertyUsage ?usageiri . ?usageiri rdf:type ?usagetypeiri; ` retrieves usagetypeiri which is stored in postgis is queried via Ontop. The usagetypeiri was not uploaded into Postgis but was processed by running OSMAgent. 

I hope this clarifies. Happy to jump into a call to discuss any other questions you may have.

Reply:
Ok.

This is highly regrettable. Can you please add a 'to do' section to the readme file, explaining the situation? That is, add a paragraph that makes it clear that 1) extracting information from IRI strings is generally unacceptable and should under no circumstances be copied or imitated, 2) the only reason this is done here is as a work-around for performance reasons, because querying the same information semantically from the KG is dramatically slower (3 s vs 30 s).

Reply:
Hi Seb,

I have added a readme file explaining the situation. Hope this works! Thank you.